Update mstest0045.md (#47612)
dotnet/docs: docs/core/testing/mstest-analyzers/mstest0045.md

## Rule description

> There is no way to gracefully abort a running thread. So, for a timeout to work, there are two ways:

Reviewer: ```suggestion
There's no way to gracefully abort a running thread. There are two ways for a timeout to work:
```

> 1. Stop observing the thread running the test. But this can be problematic in many cases and might make the remaining tests unstable due to potential race conditions. This is non-cooperative cancellation.
> 2. A cancellation is requested once the timeout is reached, and it's the test responsibility to terminate on request. This is cooperative cancellation.

Reviewer: ```suggestion
- Stop observing the thread running the test. But this can be problematic in many cases and might make the remaining tests unstable due to potential race conditions. This is *non-cooperative cancellation*.
- Request cancellation once the timeout is reached, and it's the test responsibility to terminate on request. This is *cooperative cancellation*.
```
